python_src/models/ac3rp/common.py

```diff
@@ -0,0 +1,206 @@
+    #           ]
+    #         },
+    # Get the points defining the initial part of the first driving actions for the vehicle
+    trajectory_points = driving_actions[0]["trajectory"][0]
+
+    initial_location = (trajectory_points[0][0], trajectory_points[0][1])
+
+    if len(trajectory_points) == 2:
+        initial_point = Point(trajectory_points[0][0], trajectory_points[0][1])
+        final_point = Point(trajectory_points[1][0], trajectory_points[1][1])
+    elif len(trajectory_points) == 3:
+        # Find the circle
+        initial_arc_point = Point(trajectory_points[0][0], trajectory_points[0][1])
+        middle_arc_point = Point(trajectory_points[1][0], trajectory_points[1][1])
+        final_arc_point = Point(trajectory_points[2][0], trajectory_points[2][1])
+        #
+        radius, center = find_radius_and_center(initial_arc_point, middle_arc_point, final_arc_point)
+        # Take the vector from the center to the initial point
+        the_radius_vector = LineString([center, initial_arc_point])
+        # Translate that to the origin
+        the_radius_vector = translate(the_radius_vector, xoff=-center.x, yoff=-center.y)
+
+        # We really care only about the sign to understand which of the two tangents to take
+        # Find the angle between: the vectors center-initial_point, center-final_point
+        # initialize arrays
+        A = array([initial_arc_point.x - center.x, initial_arc_point.y - center.y])
+        B = array([final_arc_point.x - center.x, final_arc_point.y - center.y])
+        # For us clockwise must be positive so we need to invert the sign
+        direction = -1.0 * copysign(1.0, cross(A, B))
+        angle_between_segments = atan2(abs(cross(A, B)), dot(A, B))
+        #
+        the_angle = degrees(direction * angle_between_segments)
+        if the_angle >= 0.0:
+            the_radius_vector = rotate(the_radius_vector, 90.0, (0, 0), use_radians=False)
+        else:
+            the_radius_vector = rotate(the_radius_vector, -90.0, (0, 0), use_radians=False)
+
+        # This should be the origin !
+        # initial_point = Point(the_radius_vector.coords[0])
+        initial_point = Point(0, 0)
+        final_point = Point(the_radius_vector.coords[-1])
+    elif len(trajectory_points) == 1:
+        return initial_location, 0
+    else:
+        raise Exception("Not enough points to compute the initial state of vehicle")
+
+    # Find the angle between: the vectors center-initial_point, center-final_point
+    # initialize arrays
+    NORTH = array([0, 1])
+    B = array([final_point.x - initial_point.x, final_point.y - initial_point.y])
+    # For us clockwise must be positive so we need to invert the sign
+    direction = copysign(1.0, cross(NORTH, B))
+    angle_between_segments = atan2(abs(cross(NORTH, B)), dot(NORTH, B))
+    #
+    intial_rotation = degrees(direction * angle_between_segments)
+
+    return initial_location, intial_rotation
+
+
+def f7(seq):
```

Comment:
What's this? Maybe you can give the method a more descriptive name.

Reply:
The method was updated! Many thanks 👍🏻